service/frontend/nexus_handler.go

```diff
@@ -461,16 +468,18 @@ func (h *nexusHandler) CancelOperation(ctx context.Context, service, operation,
 	switch t := response.GetOutcome().(type) {
 	case *matchingservice.DispatchNexusTaskResponse_HandlerError:
 		oc.metricsHandler = oc.metricsHandler.WithTags(metrics.NexusOutcomeTag("handler_error"))
+		oc.responseHeaders[failureSourceHeaderName] = failureSourceClient
 		return &nexus.HandlerError{
-			Type:    convertNexusHandlerError(nexus.HandlerErrorType(t.HandlerError.GetErrorType())),
+			Type:    nexus.HandlerErrorType(t.HandlerError.GetErrorType()),
```

Comment:
Let's make sure to convert unknown/unset types to internal errors (for both type and failure message) and log.